File: Client/Tuckshop.WebApp/src/Reporting/Reports/Example/ExampleCriteria.ts
import { NeoModel, Rules, Validation, ValueObject } from '@singularsystems/neo-core';

@NeoModel
export default class ExampleCriteria extends ValueObject {

    @Rules.Required()
    public searchString: string | null = null;

    // Client only properties / methods

    protected addBusinessRules(rules: Validation.Rules<this>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        return "Example Criteria";
    }
}

File: Server/Tuckshop.Domain/Tuckshop.Reporting/Tuckshop.Reporting/Reports/Example/ExampleCriteria.cs
namespace Tuckshop.Reporting.Reports.Example
{
  using System.ComponentModel.DataAnnotations;
  using Neo.Model;
  using Neo.Model.Validation;

  /// <summary>
  /// The example report.
  /// </summary>
  public class ExampleCriteria : CriteriaBase<ExampleCriteria>
  {
    /// <summary>
    /// Gets or sets the search string to filter by.
    /// </summary>
    [Required]
    public string? SearchString { get; set; }

    /// <inheritdoc/>
    protected override void AddBusinessRules(ValidationRules<ExampleCriteria> rules)
    {
      base.AddBusinessRules(rules);
    }
  }
}
